src/clarity-angular/popover/signpost/if-open.directive.ts

```diff
@@ -0,0 +1,112 @@
+ * The full license information can be found in LICENSE in the root directory of this project.
+ */
+import {
+    Directive,
+    TemplateRef,
+    ViewContainerRef,
+    Input,
+    OnInit, Output, EventEmitter
+} from "@angular/core";
+
+import { IfOpenService } from "./if-open.service";
+
+@Directive({
+    selector: "[clrIfOpen]"
+})
+
+/**********
+ *
+ * @class IfOpenDirective
+ *
+ * @description
+ * A structural directive that controls whether or not the associated TemplateRef is instantiated or not.
+ * It makes use of a Component instance level service: IfOpenService to maintain state between itself and the component
+ * using it in the component template.
+ *
+ */
+export class IfOpenDirective implements OnInit {
+
+    /*********
+     * @property _open
+     *
+     * @description
+     * A property that holds the open or closed state for the template implementing this directive.
+     *
+     * @type {boolean}
+     * @private
+     */
+    private _open: boolean = false;
+
+    /*********
+     * @function open
+     *
+     * @description
+     * A setter for the directive property _open.
+     *
+     * @param value
+     */
+    @Input("clrIfOpen")
+    public set open( value: boolean ) {
+        value = !!value;
+        this._open = value;
+        this.openChange.emit(value);
+    }
+
+    /**********
+     * @property openChange
+     *
+     * @description
+     * An event emitter that emits when the open property is set to allow for 2way binding when the directive is
+     * used with de-structured / de-sugared syntax.
+     *
+     * @type {EventEmitter<boolean>}
+     */
+    @Output("clrIfOpenChange") openChange: EventEmitter<boolean> = new EventEmitter<boolean>(false);
+
+    /********
+     * @function open
+     *
+     * @description
+     * A getter for the directive property _open.
+     * @returns {boolean}
+     */
+    public get open() {
+        return this._open;
+    }
+
+    constructor( private ifOpenService: IfOpenService, private template: TemplateRef<any>,
+                 private container: ViewContainerRef ) { }
+
+    /*********
+     * @function updateView
+     *
+     * @description
+     * Function that takes a boolean value and either created an embedded view for the associated ViewContainerRef or,
+     * Clears all views from the ViewContainerRef
+     * @param value
+     */
+    public updateView( value: boolean ) {
+        if ( value ) {
+            this.container.createEmbeddedView(this.template);
+        } else {
+            this.container.clear();
+        }
+    }
+
+
+    /**********
+     * @function ngOnInit
+     *
+     * @description
+     * Angular lifecycle function that subscribes to the Observable changes to the open propoerty of the
+     * IfOpenService. When changes are heard it calls updateView with the change value.
+     */
+    ngOnInit() {
+        this.ifOpenService.openedChange.subscribe((change) => {
+            this.updateView(change);
```

Comment:
I think that you should emit the event here but again we should check with @youdz.